job-service-scheduled-executor/src/main/java/com/hpe/caf/services/job/scheduled/executor/ScheduledExecutor.java

```diff
@@ -29,28 +31,43 @@
 
     public static void main(final String[] args)
     {
+        runJobs();
+    }
+
+    private static void runJobs() {
         // Create a scheduler to process scheduled tasks.
         final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
-
+    
         LOG.info("Starting Job Service Scheduled Executor service ...");
-        final Runnable task = () -> {
-            try {
-                DatabasePoller.pollDatabaseForJobsToRun();
-            } catch (final Throwable t ) {   // Catch Exceptions and Errors to prevent scheduler stoppage.
-                LOG.error("Caught exception while polling the Job Service database. Message:\n" + t.getMessage()
-                        + "StackTrace:\n" + Arrays.toString(t.getStackTrace()));
-            }
-        };
-
+        
+        final Runnable task = () -> runAvailableJobs("Auto");
+    
         //  Poll the Job Service database using the specified polling period configuration to specify how often the
         //  scheduled task is run.
         scheduler.scheduleWithFixedDelay(task, 20, ScheduledExecutorConfig.getScheduledExecutorPeriod(),
                 TimeUnit.SECONDS);
-
+    
         LOG.info("Starting task for dropping soft deleted tables ...");
         //  Execute the dropTablesTask periodically.
         scheduler.scheduleWithFixedDelay(new DropTablesTask(), 20, ScheduledExecutorConfig.getDropTablesSchedulerPeriod(),
                 TimeUnit.SECONDS);
     }
-
+    
+    /**
+     *
+     * @param origin the trigger's origin. It can be "Auto" or "Manual"
+     */
+    public static void runAvailableJobs(final String origin)
+    {
+        try {
+            final Instant start = Instant.now();
+            DatabasePoller.pollDatabaseForJobsToRun();
+            final Instant end = Instant.now();
+            LOG.debug("Total time taken to execute scheduler for {} task in ms {}", origin, Duration.between(start, end).toMillis());
```

Comment:
Consider using `isDebugEnabled()` when there is extra logic for the logging.

```suggestion
            if (LOG.isDebugEnabled()) {
                final Instant start = Instant.now();
                DatabasePoller.pollDatabaseForJobsToRun();
                final Instant end = Instant.now();
                LOG.debug("Total time taken to execute scheduler for {} task in ms {}", origin, Duration.between(start, end).toMillis());
            } else {
                DatabasePoller.pollDatabaseForJobsToRun();
            }
```

Reply:
Ok, I didn't know that one. Thx